Bump `@ledgerhq/hw-app-eth` dependency to 5.38

r-czajkowski opened the pull request:
One of the users reported a `0x6985 ledger error` which may occur in a random transaction. I was not able to reproduce this bug against ropsten and local network. Following the [btchip user recomendation](https://www.reddit.com/r/ledgerwallet/comments/j8bfng/when_trying_to_mint_a_nft_on_rarible_i_keep/g8aotp7\?utm_source\=share\&utm_medium\=web2x\&context\=3), we updated here `@ledgerhq/hw-app-eth` dependency to the `5.38`  to avoid this error in the future.

r-czajkowski committed:
Bump `@ledgerhq/hw-app-eth` dependency to 5.38

One of the users reported a `0x6985 ledger error` which may occur in a
random transaction. Following the [btchip user recomendation](
https://www.reddit.com/r/ledgerwallet/comments/j8bfng/when_trying_to_mint_a_nft_on_rarible_i_keep/g8aotp7\?utm_source\=share\&utm_medium\=web2x\&context\=3),
we updated here `@ledgerhq/hw-app-eth` dependency to the `5.38`  to
avoid this error in the future.

r-czajkowski committed:
Fix `getChainIdFromV` function

The `getChainIdFromV` function takes `v` param in `Uint8Array`. To be
able to parse this param to `int` we should call `v.toString("hex")` as
a first step otherwise, the `parseInt` function won't be able to parse
correctly to `int` and return `undefined`.
Also added a jsdoc for this function insted of a comment.

r-czajkowski committed:
Ledger integration for chain id > 255

The comments that the ledger eth app returns the lower 2 bytes are no
longer valid. Currently the eth app returns correct `v` param for
chainID > 255 according to the EIP-155. So we don't need to `signedV &
0xff === ledgerSignedV`. Updating the `@ledgerhq/hw-app-eth` dependency
to the latest version (v5.38) did the trick. There are no effects on
mainnet or ropsten build of the KEEP token dashboard since the chain id
for these networks are 1 and 3.

michalsmiarowski commented on solidity/dashboard/src/connectors/utils.js:
```diff
@@ -14,11 +14,18 @@ export const getEthereumTxObj = (txData, chainId) => {
+ * [1]: https://github.com/ethereum/EIPs/blob/master/EIPS/eip-155.md
+ * Formula: v = CHAIN_ID * 2 + 35 => CHAIN_ID = (v - 35) / 2
+ *
+ * @param {Uint8Array} vInHex v param in hex.
```
I would personally add hyphen between the name of the parameter and its description. As stated in [JSDoc](https://jsdoc.app/tags-param.html#names-types-and-descriptions) the hyphen is not necessary but it makes it more readable in my opinion. I'll leave it up to you to decide.

Shadowfiend replied:
I'm actually puzzled, is this a uint8array or a string (which is what "in hex" implies to me)?

On the subject of the dash, I think the dash becomes less clarifying if the parameter description is a full sentence (because it typically starts with a capital letter). Eg here, "The signature `v` value to extract the current chain from." or similar.

r-czajkowski committed:
Update package-lock.json

This reverts the `package-lock.json` from commit 27d503965 and updates
only necessary packages in `package-lock` after updating the
`@ledgerhq/hw-app-eth` package.

r-czajkowski committed:
Update `getChainIdFromV` funciton

Renamed `vInHex` -> `v`. The `in hex` implies that is a string, but
actually is `Uint8Array`.

pdyraga commented:
Is it still relevant or are we good to close it?
